Libraries/fishhook/fishhook.c

```diff
@@ -103,7 +103,9 @@ static void perform_rebinding_with_section(struct rebindings_entry *rebindings,
               indirect_symbol_bindings[i] != cur->rebindings[j].replacement) {
             *(cur->rebindings[j].replaced) = indirect_symbol_bindings[i];
           }
-          indirect_symbol_bindings[i] = cur->rebindings[j].replacement;
+          if (i < ( sizeof(indirect_symbol_bindings) / sizeof(indirect_symbol_bindings[0]))) {
+              indirect_symbol_bindings[i] = cur->rebindings[j].replacement;
+          }
```

Comment:
Why `indirect_symbol_bindings` would out of bounds? If we bypass the assignment, hook would be invalid IMO.